client/python/lib/test/test_grpc_requests.py

```diff
@@ -75,3 +78,52 @@ def test_make_metadata_request_invalid(mocker, name, version, expected_exception
     
     assert str(e_info.value) == expected_message
     mock_method.assert_called_once()
+
+@pytest.mark.parametrize("inputs, expected_proto, name, version", PREDICT_REQUEST_VALID)
+def test_make_predict_request_valid(inputs, expected_proto, name, version):
+    model_predict_request = make_predict_request(inputs ,name, version)
+    
+    assert isinstance(model_predict_request, GrpcPredictRequest)
+    assert model_predict_request.model_name == name
+    assert model_predict_request.model_version == version
+    assert model_predict_request.inputs == inputs
+    assert isinstance(model_predict_request.raw_request, PredictRequest)
+    assert model_predict_request.raw_request.model_spec.name == name
+    assert model_predict_request.raw_request.model_spec.version.value == version
+    assert len(inputs.keys()) == len(list(model_predict_request.raw_request.inputs.keys()))
+    for key, value in inputs.items():
+        assert isinstance(model_predict_request.raw_request.inputs[key], TensorProto)
+        if isinstance(value, TensorProto):
+            assert value == model_predict_request.raw_request.inputs[key]
+        else:
+            np_values = array(value)
+            if expected_proto[key]["field"] == "tensor_content":
+                assert model_predict_request.raw_request.inputs[key].__getattribute__(expected_proto[key]['field']) == np_values.tobytes()
+            elif expected_proto[key]["field"] == "string_val":
+                assert model_predict_request.raw_request.inputs[key].__getattribute__(expected_proto[key]['field']) == [value]
+            else:
+                assert model_predict_request.raw_request.inputs[key].__getattribute__(expected_proto[key]['field']) == np_values
```

Comment:
Can we avoid this if.. else.. block? Is it possible to put expected values into parametrization? And then go with:
`assert model_predict_request.raw_request.inputs[key].__getattribute__(expected_proto[key]['field']) == expected_proto[key]['value']`